features/steps/shell.py

```diff
@@ -64,6 +64,15 @@ def when_i_run_command(
 ):
     command = process_template_vars(context, command)
     prefix = get_command_prefix_for_user_spec(user_spec)
+
+    if context.pro_config.collect_coverage:
+        if command.split()[0] == "pro":
+            split_command = command.split()
+            split_command[
+                0
+            ] = "python3 -m coverage run --source=/usr/lib/python3/dist-packages/uaclient /usr/bin/pro"  # noqa: E501
+            command = " ".join(split_command)
+
```

Comment:
Different syntax suggestion, a little clearer to me and avoiding the noqa.
```suggestion
        coverage_command = (
            "python3 -m coverage run "
            "--source=/usr/lib/python3/dist-packages/uaclient /usr/bin/pro"
        )
        split_command = command.split()
        if split_command[0] == "pro":
            command = " ".join([coverage_command] + split_command[1:])

```

Reply:
Yeah, this looks better, thanks!

Reply:
Updated @renanrodrigo